### PR TITLE
SourceMap support for Kernel#eval

### PR DESCRIPTION
**Pull request by janbiedermann**
image

Kinda working in Chrome Browsers

### CODE REVIEW

**Comment by janbiedermann** [comment]
image
Also works in Firefox 112

File: opal/corelib/string/encoding.rb [hunk 1]
```diff
@@ -103,6 +103,7 @@ def bytesize(*)
 
   class ::EncodingError < ::StandardError; end
   class ::CompatibilityError < ::EncodingError; end
+  class UndefinedConversionError < ::EncodingError; end
```
Comment (elia) [inline opal/corelib/string/encoding.rb:106]: Can you please add the double column here? It makes the lookup faster.

**Review by elia — APPROVED** [review]
This is great, I just left a comment for a quick fix and we good to go 👍👍👍

File: opal/corelib/string/encoding.rb (hunk 1 above)
Comment (janbiedermann) [reply opal/corelib/string/encoding.rb:106]: That wouldn't be correct, it must be Encoding::UndefinedConversionError, see https://github.com/opal/opal/blob/89661b5226079ebf20230412c7c3360108d8a3ae/lib/opal/source_map/map.rb#L14

Comment (elia) [reply opal/corelib/string/encoding.rb:106]: Oh, right!

**Pull request merged**